wave_calculator.gd

```diff
@@ -61,15 +92,19 @@ func _is_valid_score(score: int) -> bool:
 
 func _get_difficulty(wave_number: int) -> int:
 	# Diffuclty increases linearly, which is boring, should have some more interesting curve
-	return 5 + (wave_number * 4)
+	return 5 + log(wave_number + 1) + wave_number
 
 
 func _get_max_enemy_score(difficulty: int) -> int:
 	@warning_ignore("integer_division")
-	return max(ceil(difficulty / 10), 1)
+	return max(ceil(difficulty / 6), 1)
 
 
 func _sum(array: Array[int]) -> int:
 	if array.is_empty():
 		return 0
 	return array.reduce(func(x, y): return x + y)
+
+
+func _filter_lte(limit: int) -> Callable:
```

Comment:
it's redundant that this is called filter, just `lte` or even better `lessThan` would be better.

Reply:
changed this to `_lessThanEqualTo(limit: int) -> Callable:`